### PR TITLE
Perform status updates / note edits in the main frame rather than the iframe and refresh the page on success

### PR DESCRIPTION
**Pull request by clouser2**
### Description

As part of #3268, we'd like to show the status in the application list, which is part of the main page. Previously, any status updates / note edits were performed within the IFrame. As such, the list of applications on the side would not update (and reflect the new status).

This implements the approach described in https://github.com/civiform/civiform/issues/3268#issuecomment-1239575736 and https://github.com/civiform/civiform/issues/3268#issuecomment-1239858521.

After submission, we'll add the status to the application list since it'll be properly updated at that point.

## Release notes:

Program admins updating application status and notes will now see a full page refresh instead of only seeing the application view be updated.

### Checklist

- [X] Added the correct label: < feature | bug | dependencies | infrastructure | ignore-for-release >
- [X] Created tests which fail without the change (if possible)
- [ ] Extended the README / documentation, if necessary

### Issue(s)

#3268

### CODE REVIEW

**Review by shanemc-goog — CHANGES_REQUESTED** [review]

File: browser-test/src/admin_application_statuses.test.ts [hunk 1]
```diff
@@ -122,6 +122,13 @@ describe('view program statuses', () => {
         await adminPrograms.confirmStatusUpdateModal(modal)
         expect(await adminPrograms.getStatusOption()).toBe(noEmailStatusName)
         await adminPrograms.expectUpdateStatusToast()
+
+        // Confirm that the application is shown after reloading the page.
+        const applicationText = await adminPrograms
+          .applicationFrameLocator()
+          .locator('#application-view')
+          .innerText()
+        expect(applicationText).toContain('Guest')
```
[the remark this answers is not in the document]
Comment (clouser2) [reply browser-test/src/admin_application_statuses.test.ts:131]: Good call. Also asserted some of the relevant answer text (similar to what we were doing when asserting downloaded JSON/CSV content.

File: server/app/assets/javascripts/admin_applications.ts [hunk 2]
```diff
@@ -72,6 +92,141 @@ class AdminApplications {
...
+    formEl.action = action
+    // Retrieve the CSRF token from the page.
+    formEl.appendChild(
+      this._assertNotNull(
+        document.querySelector('input[name=csrfToken]'),
+        'csrf token',
+      ),
+    )
+    inputs.forEach(({inputName, inputValue}) => {
+      const elementType = inputValue.includes('\n') ? 'textarea' : 'input'
```
[the remark this answers is not in the document]
Comment (shanemc-goog) [reply server/app/assets/javascripts/admin_applications.ts:202]: oh, this won't affect the display ? I was thinking a one line email body would be in a 1 line input and not easily expandable.

File: server/app/views/admin/programs/ProgramApplicationListView.java [hunk 3]
```diff
@@ -160,7 +158,7 @@ private FormTag renderSearchForm(
                     /* fromDate= */ Optional.empty(),
                     /* untilDate= */ Optional.empty(),
                     /* applicationStatus= */ Optional.empty(),
-                    /* selectedApplication=*/ Optional.empty())
+                    /* selectedApplicationUri=*/ Optional.empty())
```
Comment (shanemc-goog) [inline server/app/views/admin/programs/ProgramApplicationListView.java:161]:
```suggestion
                    /* selectedApplicationUri= */ Optional.empty())
```

**Review by shanemc-goog — APPROVED** [review]

File: server/app/assets/javascripts/admin_applications.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (clouser2) [reply server/app/assets/javascripts/admin_applications.ts:202]: The form created here is never displayed. It's merely created via JS and submitted immediately, which triggers a page refresh.

**Pull request merged**